java/ql/src/experimental/Security/BigDecimalDOS/BigDecimalDOS.ql

```diff
@@ -0,0 +1,38 @@
+/**
+ * @id java/BigDecimalDOS
+ * @name Java-BigDecimal-DOS-Vulnerability
+ * @description When user-controllable data is pass to the relevant Methods in BigDecimal, it may cause DOS issues when computing resources are limited. This issue is common in business scenarios such as e-commerce platforms.
```

Comment:
```suggestion
 * @description When user-controllable data is passed to the relevant methods in BigDecimal, it may cause DOS issues when computing resources are limited. This issue is common in business scenarios such as e-commerce platforms.
```
Maybe "This issue is common in business scenarios such as e-commerce platforms." can be omitted here.